### PR TITLE
Build2020

### PR DESCRIPTION
**Pull request by arturl**
This adds new Bot as an agent sample for LivePerson

### CODE REVIEW

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            else
+            {
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                if (userText.Contains("disneyland"))
+                {
+                    await turnContext.SendActivityAsync("Sorry I cannot answer your questoin but an agent will be with you shortly");
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:62]: spelling. "Question"

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                await turnContext.SendActivityAsync(evnt);
+            }
+            else
+            {
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                if (userText.Contains("disneyland"))
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:60]: I'm not sure the context here; is case + culture comparison needed?

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    activity.ChannelData = JObject.Parse(channelData);
+
+                    await turnContext.SendActivityAsync(activity);
+                    return;
+                }
+                else
+                {
+                    foreach (var country in Capitals.Keys)
+                    {
+                        if (userText.Contains(country.ToLower()))
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:84]: Do you want a culture aware ToLower()? I don't think the thread culture is ever being set, so I suspect not.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    }
+                }
+
+                await turnContext.SendActivityAsync(MessageFactory.Text(replyText, replyText), cancellationToken);
+            }
+        }
+
+        protected override async Task OnEventAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if(turnContext.Activity.Name == "handoff.status")
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:97]: Consider adding a constant for this.

Comment (cleemullins) [reply experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:97]: Also consider extending the base class that looks at activity types and then raises events via virtual method. 

There's nothing wrong with this code, it's just a bit different than how most of our samples and docs are structured.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            {
+                var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+                var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+                Activity replyActivity;
+                var state = (turnContext.Activity.Value as JObject)?.Value<string>("state");
+                if (state == "typing")
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:104]: Same comment around constants.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Directory.Build.props [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<Project>
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Directory.Build.props:1]: Is this file needed? Seems like it's not being used and could be removed.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/EchoBot.runtimeconfig.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "runtimeOptions": {
+    "tfm": "netcoreapp2.1",
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/EchoBot.runtimeconfig.json:3]: I would *strongly* consider replatforming this to Net Core 3.1. Net core 2.1 is already in "Long Term Support" and is really not recommended for new projects at this point. 
https://dotnet.microsoft.com/platform/support/policy/dotnet-core

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/EchoBot.runtimeconfig.json [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    "framework": {
+      "name": "Microsoft.AspNetCore.App",
+      "version": "2.1.1"
+    },
+    "configProperties": {
+      "System.GC.Server": true
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/EchoBot.runtimeconfig.json:9]: Is this something we want? I don't believe overriding the defaults value here is a good idea. We should just leave the ASP.Net defaults "as-is".

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentAdapter.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentAdapter.cs:1]: Copyright headers

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentAdapter.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentAdapter.cs:4]: Remove and Sort using statements. I don't think most of these are needed.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj [hunk 11]
```diff
@@ -0,0 +1,33 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.1</TargetFramework>
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj:4]: Duplicate from the other file; same comment applies. Let's get rid of .Net core 2.1 for new stuff; new stuff needs to be .Net Core 3.1.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  <ItemGroup>
+    <Compile Remove="LivePersonConnector\**" />
+    <Content Remove="LivePersonConnector\**" />
+    <EmbeddedResource Remove="LivePersonConnector\**" />
+    <None Remove="LivePersonConnector\**" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.Bot.Builder.Integration.AspNet.Core" Version="4.8.0" />
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj:17]: 4.9.1

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="2.1.10" />
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj:18]: Is this package needed? I don't think it is.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Update="appsettings.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  
+  <Import Project="PostDeployScripts\IncludeSources.targets" Condition="Exists('PostDeployScripts\IncludeSources.targets')" />
+  <Import Project="..\PostDeployScripts\IncludeSources.targets" Condition="Exists('..\PostDeployScripts\IncludeSources.targets')" />
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.csproj:28]: I don't think this would work on a Mac/Linux environment due to the "\". I may be mistaken, but it would need to be checked.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.sln [hunk 15]
```diff
@@ -0,0 +1,24 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/LivePersonAgentBot.sln:1]: We don't typically check solution files in, as all that's needed is the .csproj file. Consider removing this file.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs [hunk 16]
```diff
@@ -0,0 +1,75 @@
+﻿using System;
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs:1]: Copyright notice.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+using Microsoft.Bot.Builder;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+using LivePersonAgentBot.Bots;
+using System.Net.Http;
+using Newtonsoft.Json;
+using System.Text;
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs:11]: Remove and sort usings. There seems to be many not-necessary using statments, and their order seems random.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+using System.Collections.Concurrent;
+using Newtonsoft.Json.Linq;
+
+namespace LivePersonAgentBot
+{
+    public class HandoffMiddleware : Microsoft.Bot.Builder.IMiddleware
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs:17]: The very first using is "Microsoft.Bot.Builder", so I'm not clear on why the full namespace is here on IMiddleware. 

Consider running the standard .Net linting / rosyln analyzers over this code.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+    {
+        public HandoffMiddleware()
+        {
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default)
+        {
+            turnContext.OnSendActivities(async (sendTurnContext, activities, nextSend) =>
+            {
+                Activity handOffEventActivity = activities.Where((activity) => activity.Type == ActivityTypes.Event && activity.Name == HandoffEventNames.InitiateHandoff).FirstOrDefault();
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs:27]: This statement is quite complex, involving LINQ, delegates, operator precedence, and default values. 

Consider simplifying for clarity. 

Also, there's no guarantee that activities won't be null, as it wasn't checked - so this may all throw a NRE.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+                if (handOffEventActivity != null)
+                {
+                    var context = handOffEventActivity.Value as JObject;
+
+                    var skill = context?.Value<string>("Skill");
+                    if(String.IsNullOrEmpty(skill))
+                    {
+                        skill = "default";
+                    }
+#if false // using string literal - works but more verbose
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Middleware/HandoffMiddleware.cs:38]: Consider just choosing one or the other approaches. 

My personal choice would be to move this to an embedded resource file, and just pull it in. that's easier to edit, clear, and scalable across multiple different versions of this.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/IncludeSources.targets [hunk 21]
```diff
@@ -0,0 +1,15 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/IncludeSources.targets:1]: It's been a while since I looked at an ASP.Net application that did this. Is this still necessary or can this file be removed? I don't think it's needed.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/githubProject.json.template [hunk 22]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/githubProject.json.template:1]: Is this right?

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/mergeSettings.js [hunk 23]
```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/mergeSettings.js:1]: Copyright

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/mergeSettings.js [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+const path = require('path');
+const appSettingPath = path.resolve('./appsettings.json');
+const envPath = path.resolve('./.env');
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/mergeSettings.js:4]: This file is quite complex. None of our existing samples do anything like this. Is this something we need here?

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/prepareSrc.cmd [hunk 25]
```diff
@@ -0,0 +1,89 @@
+@echo off
+setlocal
+SET password=%1
+SET exportSettings=%2
+SET repoName=srcRepo
+SET repoUrl=file:///%HOMEDRIVE:~0,1%/%HOMEPATH:~1%/site/%repoName%
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/prepareSrc.cmd:6]: This won't work on Mac or Linux. 

Please remove and consider an alternate solution.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/runGulp.cmd [hunk 26]
```diff
@@ -0,0 +1,23 @@
+@echo off
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/PostDeployScripts/runGulp.cmd:1]: I think I'm not understanding what these CMD files are for. Again ,they all won't run on Mac/Linux which is something we need our samples to do.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/README.md [hunk 27]
```diff
@@ -0,0 +1,4 @@
+# LivePerson Agent Bot
+
+This sample demonstrates how to connect any Bot Framework channel to the LivePerson agent hub. The bot uses the new Bot Framework Handoff APIs.
+
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/README.md:4]: Consider significantly expanding on this.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+
+            var storage = new MemoryStorage();
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs:28]: We've gone back-and-forth over this line. Do you believe users will use this sample "as-is"? If so, a comment here describing that this sample is all transient (state is not durable) is needed here or in the readme.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs:26]: As you migrate this to .Net Core 3.1, this need to change. All of our R9 samples are updates already in master, so you can just cut/paste.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/UserProfile.cs [hunk 30]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace LivePersonAgentBot.Bots
+{
+    internal class UserProfile
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/UserProfile.cs:6]: This class isn't referenced anywhere and seems like it could be deleted.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/build.cmd [hunk 31]
```diff
@@ -0,0 +1,18 @@
+@echo off
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/build.cmd:1]: Windows specific again. NEeds to be cross platform.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs (hunk 28 above)
Comment (johnataylor) [reply experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Startup.cs:28]: I would definitely add a comment along these lines "MemoryStorage is designed for testing purposes"

**Comment by arturl** [comment]
All fixed, thanks.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 32]
```diff
@@ -0,0 +1,109 @@
...
+            ["Poland"] = "Warsaw",
+            ["Germany"] = "Hamburg" // not really
+        };
+
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+            var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+            var userText = turnContext.Activity.Text.ToLower();
```
Comment (carlosscastro) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:37]: Suggest `ToLowerInvariant()` for better multi-culture support

```suggestion
            var userText = turnContext.Activity.Text.ToLowerInvariant();
```

Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs]: I don't understand why there's a .ToLower() here at all. Why normalize this to lower-case?

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 33]
```diff
@@ -0,0 +1,109 @@
...
+                string replyText = $"Sorry, I cannot help you.";
+                if (userText == "hi")
+                {
+                    replyText = "Hello!";
+                }
+                else
+                {
+                    foreach (var country in Capitals.Keys)
+                    {
+                        if (userText.Contains(country.ToLower()))
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:61]: I would avoid the .ToLower() normalizing (which are incorrect, as the Activity's user text and this thread may have different locales, meaning the comparison isn't valid), and instead use a case insensitive or ordinal based compare.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs [hunk 34]
```diff
@@ -0,0 +1,109 @@
...
+            if(turnContext.Activity.Name == HandoffEventNames.HandoffStatus)
+            {
+                var conversationStateAccessors = _conversationState.CreateProperty<LoggingConversationData>(nameof(LoggingConversationData));
+                var conversationData = await conversationStateAccessors.GetAsync(turnContext, () => new LoggingConversationData());
+
+                Activity replyActivity;
+                var state = (turnContext.Activity.Value as JObject)?.Value<string>("state");
+                if (state == HandoffStates.Typing)
+                {
+                    replyActivity = new Activity
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:83]: consider pulling through the locale of the original activity.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Controllers/BotController.cs [hunk 35]
```diff
@@ -0,0 +1,35 @@
...
+namespace LivePersonAgentBot.Controllers
+{
+    // This ASP Controller is created to handle a request. Dependency Injection will provide the Adapter and IBot
+    // implementation at runtime. Multiple different IBot implementations running at different endpoints can be
+    // achieved by specifying a more specific type for the bot constructor argument.
+    [Route("api/messages")]
+    [ApiController]
+    public class BotController : ControllerBase
+    {
+        private readonly IBotFrameworkHttpAdapter Adapter;
```
Comment (cleemullins) [inline experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Controllers/BotController.cs:18]: our current coding standards for this are "_adapter". We're starting to enforce that via the Analyzers in the main repos, but if it's easy I would update this code as well.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs (hunk 32 above)
Comment (arturl) [reply experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs]: This way user can ask for an agent, an Agent or an AGENT (if angry)

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Controllers/BotController.cs (hunk 35 above)
Comment (arturl) [reply experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Controllers/BotController.cs:18]: I'll address this consistently in all projects in the new iteration -- more changes coming later.

File: experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs (hunk 34 above)
Comment (arturl) [reply experimental/handoff-library/csharp_dotnetcore/samples/LivePersonAgentBot/Bots/LivePersonAgentBot.cs:83]: will consider in the future
